tests/src/fit_cov.cpp

```diff
@@ -0,0 +1,237 @@
+/*
+ Copyright (C) 2014 Nicolas Mellado <nmellado0@gmail.com>
+
+ This Source Code Form is subject to the terms of the Mozilla Public
+ License, v. 2.0. If a copy of the MPL was not distributed with this
+ file, You can obtain one at http://mozilla.org/MPL/2.0/.
+*/
+
+
+/*!
+ \file test/src/fit_cov.cpp
+ \brief Test validity of the covariance fitting procedures wrt to standard algorithm
+ */
+
+#include "../common/testing.h"
+#include "../common/testUtils.h"
+
+#include <Ponca/src/Fitting/basket.h>
+#include <Ponca/src/Fitting/defines.h>
+#include <Ponca/src/Fitting/mean.h>
+#include <Ponca/src/Fitting/covarianceFit.h>
+#include <Ponca/src/Fitting/weightFunc.h>
+#include <Ponca/src/Fitting/weightKernel.h>
+
+#include <vector>
+
+using namespace std;
+using namespace Ponca;
+
+
+/// Class that perform the covariance fit using standard two-passes procedure
+template < class DataPoint, class _WFunctor, typename T>
+class CovarianceFitTwoPassesBase : public T
+{
+    PONCA_FITTING_DECLARE_DEFAULT_TYPES
+
+protected:
+    enum
+    {
+        Check = Base::PROVIDES_MEAN_POSITION,
+        PROVIDES_POSITION_COVARIANCE
+    };
+
+public:
+    using MatrixType = typename DataPoint::MatrixType; /*!< \brief Alias to matrix type*/
+    /*! \brief Solver used to analyse the covariance matrix*/
+    using Solver = Eigen::SelfAdjointEigenSolver<MatrixType>;
+
+protected:
+    // computation data
+    MatrixType m_cov {MatrixType::Zero()};     /*!< \brief Covariance matrix */
+    Solver m_solver;  /*!<\brief Solver used to analyse the covariance matrix */
+    bool m_barycenterReady {false};
+    VectorType m_barycenter;
+    Scalar sumW;
+
+public:
+    PONCA_EXPLICIT_CAST_OPERATORS(CovarianceFitTwoPassesBase,covarianceFitTwoPasses)
+    PONCA_FITTING_DECLARE_INIT_ADD_FINALIZE
+
+    /*! \brief Reading access to the Solver used to analyse the covariance matrix */
+    PONCA_MULTIARCH inline const Solver& solver() const { return m_solver; }
+};
+
+template < class DataPoint, class _WFunctor, typename T>
+void
+CovarianceFitTwoPassesBase<DataPoint, _WFunctor, T>::init(const CovarianceFitTwoPassesBase<DataPoint, _WFunctor, T>::VectorType& _evalPos)
```

Comment:
Is the type prefix necessary? `VectorType` instead of `CovarianceFitTwoPassesBase<DataPoint, _WFunctor, T>::VectorType` would be more readable

Reply:
Testing with the CI, will resolve if it all goes well, thx